drivers/storage/portworx/component/telemetry.go

```diff
@@ -697,6 +711,46 @@ func (t *telemetry) createConfigMap(
 	)
 }
 
+func (t *telemetry) reconcileCCMProxyConfigMap(
+	cluster *corev1.StorageCluster,
+	ownerRef *metav1.OwnerReference,
+) error {
+	proxy := pxutil.GetPxProxyEnvVarValue(context.TODO(), cluster)
+	// Delete the existing config map if portworx proxy is empty
+	if proxy == "" {
+		existingConfigMap := &v1.ConfigMap{}
+		err := t.k8sClient.Get(
+			context.TODO(),
+			types.NamespacedName{
+				Name:      TelemetryCCMProxyConfigMapName,
+				Namespace: cluster.Namespace,
+			},
+			existingConfigMap,
+		)
+		if errors.IsNotFound(err) {
+			return nil
+		} else if err != nil {
+			return err
+		}
```

Comment:
We don't need this. The DeleteConfigMap does the same thing. We can directly call `DeleteConfigMap` if proxy is nil.